go/libraries/doltcore/dbfactory/dirtodbname.go

```diff
@@ -15,16 +15,20 @@
 package dbfactory
 
 import (
+	"os"
 	"strings"
 	"unicode"
+
+	"github.com/dolthub/dolt/go/libraries/doltcore/dconfig"
 )
 
 // DirToDBName takes the physical directory name, |dirName|, and replaces any unsupported characters to create a
-// valid logical database name. For example, hyphens and spaces are replaced with underscores.
+// valid logical database name. For example, spaces are replaced with underscores.
 func DirToDBName(dirName string) string {
+	var translateHyphensToUnderscores = os.Getenv(dconfig.EnvDisableDatabaseRenaming) != ""
```

Comment:
This logic is a little hard to parse... If `DOLT_DISABLE_DATABASE_RENAMING` is set to any value, then we translate hyphens to underscores? That seems backwards, right? Doesn't this mean if the user sets `DOLT_DISABLE_DATABASE_RENAMING` that we **will** rename the database? 

I think we need a clearer env var name here. Sorry if this was a name I brainstormed with you earlier. How about `DOLT_DBNAME_REPLACE_HYPHENS` or something similar? That seems more accurate in a couple of ways (e.g. we're still doing some db renaming after all) and it also avoids the negative of `DISABLE` which is hard to grok when the new change is to stop doing something (i.e. feels like a double negative to disable stopping doing something).